src/Orleans.Serialization/Codecs/ByteArrayCodec.cs

```diff
@@ -289,4 +291,53 @@ public static Memory<byte> DeepCopy(Memory<byte> input, CopyContext copyContext)
             return input.ToArray();
         }
     }
+
+    /// <summary>
+    /// Serializer for <see cref="PooledBuffer"/> instances.
+    /// </summary>
+    [RegisterSerializer]
+    public sealed class PooledBufferCodec : IValueSerializer<PooledBuffer>
+    {
+        public void Serialize<TBufferWriter>(ref Writer<TBufferWriter> writer, scoped ref PooledBuffer value) where TBufferWriter : IBufferWriter<byte>
+        {
+            writer.WriteVarUInt32((uint)value.Length);
+            foreach (var segment in value)
+            {
+                writer.Write(segment);
+            }
+
+            // Dispose of the value after sending it.
+            // PooledBuffer is special in this sense.
+            // Senders must not use the value after sending.
+            // Receivers must dispose of the value after use.
+            value.Reset();
+            value = default;
+        }
+
+        public void Deserialize<TInput>(ref Reader<TInput> reader, scoped ref PooledBuffer value)
+        {
+            const int MaxSpanLength = 4096;
```

Comment:
Why such a small buffer size? Wouldn't something like 1MB (or even bigger) be more appropriate? There's no reason to prefer small buffers when they are pooled.

Reply:
Each buffer is owned by at most one instance at a time, so if the buffers are individually large and there are many instances, then the memory footprint of the application may become unnecessarily large. Ideally, this should be changed to have adaptive buffer sizes (eg powers of 2 from 256b to 1MB), maybe with an initial size hint.